// vls/dist/modes/style/sass/sassLanguageMode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const vscode_css_languageservice_1 = require("vscode-css-languageservice");
const sass_formatter_1 = require("sass-formatter");
class SassLanguageMode {
    constructor() {
        this.config = {};
    }
    getId() {
        return 'sass';
    }
    configure(c) {
        this.config = c;
    }
    format(document, range, formattingOptions) {
        const sassConfig = {
            convert: true,
            deleteEmptyRows: true,
            deleteWhitespace: true,
            debug: false,
            insertSpaces: formattingOptions.insertSpaces,
            tabSize: formattingOptions.tabSize,
            setPropertySpace: true
        };
        Object.assign(sassConfig, this.config.sass.format);
        if (this.config.vetur.format.defaultFormatter.sass === 'sass-formatter') {
            return [vscode_css_languageservice_1.TextEdit.replace(range, sass_formatter_1.SassFormatter.Format(document.getText(range), sassConfig))];
        }
        return [];
    }
    onDocumentRemoved(document) { }
    dispose() { }
}
exports.SassLanguageMode = SassLanguageMode;
